core/src/main/java/org/apache/struts2/dispatcher/DefaultDispatcherErrorHandler.java

```diff
@@ -97,6 +97,10 @@ protected void sendErrorResponse(HttpServletRequest request, HttpServletResponse
             response.sendError(code, e.getMessage());
         } catch (IOException e1) {
             // we're already sending an error, not much else we can do if more stuff breaks
+            LOG.warn("Unable to send error response, code: {}! (IOException): {}", code, e1.toString());
+        } catch (IllegalStateException ise) {
+            // Log illegalstate instead of passing unrecoverable exception to calling thread
+            LOG.warn("Unable to send error response, code: {}! isCommited: {}. (IllegalStateException): {}", code, response.isCommitted(), ise.toString());
```

Comment:
You don't have to use `.toString()`, Log4j2 will do it for you, also this is in opposite to lines below :)

Reply:
Don't put `{}` for the exception.
```suggestion
LOG.warn("Unable to send error response, code: {}; isCommited: {}", code, response.isCommitted(), ise);
```

Reply:
Hi.

Performing the `toString()` calls was an intentional attempt to avoid Log4j2 converting it into a stacktrace (as opposed to the DevMode flow with the comment "  // Stacktrace with DevMode").  I wasn't sure if the stacktrace at this point would provide a lot of benefit (as opposed to when DevMode is true), so opted to force the short-form.

If both of you prefer just outputting the whole stacktrace all the time, I can do that instead, and make the logging symmetric in both methods.  I suppose since the `DefaultDispatchErrorHandler` can be replaced by a developer who wants custom behaviour, the default implementation can just dump the full stacktrace (since they can change it if they want or need to).

There are configuration options in Log4j2 to configure `throwable` output formatting but it only seems to be possible at the appender level (which would require some configuration gymnastics to achieve what forcing the `.toSring()` attempts here ... if it's even possible).  If anyone knows a Log4j2 configuration that could modify the output only for a specific class/method combination please feel free to share.  :smile:

Reply:
Oh, nice, I didn't know this :)